### PR TITLE
Adding Documentation for ECMA-402

### PR DESCRIPTION
**Pull request by mganandraj**
## Summary

This PR adds documentation on our Intl/ECMA402 implementation for Android platform. The contents are essentially a subset of our pre-existing docx formatted documentation [here](https://github.com/facebook/hermes/raw/master/lib/Platform/Intl/java/com/facebook/hermes/intl/HermesIntlAPIsImplementationNotes.docx)

## Test Plan

Documentation change only.

### CODE REVIEW

File: doc/intl.md [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+
+## Excluded
+
+- [formatMatcher](https://tc39.es/ecma402/#sec-basicformatmatcher) property in Intl.DateTimeFormat.
+
+## Known Issues
+
+### Android 11
+
+- The keys of the object returned by &#39;resolvedOptions&#39; function in all Intl services are not deterministically ordered as prescribed by spec.
```
Comment (dulinriley) [inline doc/intl.md:75]: Fix the HTML escape here, it should probably use backticks directly

File: doc/intl.md [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+- Intl.NumberFormat implementation has some rough edges in supporting the following properties,
+  - style: 'unit'
+  - notation: 'compact'
+  - signDisplay
+  - currencyFormat
+
+### Android 10 (SDK<30) and older
+
+- NumberFormat: Scientific notation formatting has issues on some cases such as Infinity (e.g. Expected SameValue(«-∞E0», «-∞») to be true).
+- NumberFormat: Compact notation formatToParts doesn&#39;t identify unit, hence we report unit as &#39;literal&#39; (Compact short: 987654321: parts[1].type Expected SameValue(«literal», «compact») to be true)
```
Comment (dulinriley) [inline doc/intl.md:86]: Fixup the HTML escape here as well

File: doc/intl.md [hunk 3]
```diff
@@ -0,0 +1,192 @@
+# Overview
```
Comment (dulinriley) [inline doc/intl.md:1]: Before this, can you add an ID and title for docusaurus?
The "id" is used in hermes/website/sidebars.json as a key, and "title" is displayed as the overarching title.

Make sure to also add to the sidebars.json file so this will be displayed on our website.
```suggestion
---
id: intl
title: Intl Library
---
# Overview
```

**Review by dulinriley — CHANGES_REQUESTED** [review]
Please fix up the HTML escapes.

After you add stuff to `hermes/website/sidebars.json`, make sure to test the website locally:
```
cd hermes/website
yarn install
yarn run docusaurs start
```

I think npm commands also work, but watch out for their lock files being dissimilar.

File: doc/intl.md (hunk 3 above)
Comment (Huxpro) [reply doc/intl.md:1]: maybe `title: Internationalization APIs`

File: doc/intl.md [hunk 4]
```diff
@@ -0,0 +1,192 @@
+# Overview
+
+This document describes the current state of Android implementation of Intl APIs(formally ECMA-402 specification) in Hermes JavaScript engine. ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and typically adds new capabilities typically as,
```
Comment (Huxpro) [inline doc/intl.md:3]:
```suggestion
This document describes the current state of Android implementation of [the ECMAScript Internationalization API Specification](https://tc39.es/ecma402/) (ECMA-402, or `Intl`) in Hermes JavaScript engine. ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and typically adds new capabilities typically as,

```

We should avoid the "Intl". Either go with `Intl` (note that it's formatted as code) or "Internationalization".

File: doc/intl.md [hunk 5]
```diff
@@ -0,0 +1,192 @@
+# Overview
+
+This document describes the current state of Android implementation of Intl APIs(formally ECMA-402 specification) in Hermes JavaScript engine. ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and typically adds new capabilities typically as,
+- New Intl service constructors (for e.g. Intl.Collator, Intl.NumberFormat etc...) or extending existing ones by accepting more parameters
```
Comment (Huxpro) [inline doc/intl.md:4]:
```suggestion
- New `Intl` service constructors (for e.g. `Intl.Collator`, `Intl.NumberFormat`, etc.) or extending existing ones by accepting more parameters.

```

Same to other places XD

File: doc/intl.md [hunk 6]
```diff
@@ -0,0 +1,192 @@
+# Overview
+
+This document describes the current state of Android implementation of Intl APIs(formally ECMA-402 specification) in Hermes JavaScript engine. ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and typically adds new capabilities typically as,
+- New Intl service constructors (for e.g. Intl.Collator, Intl.NumberFormat etc...) or extending existing ones by accepting more parameters
+- New functions or properties in the Intl object prototypes (for e.g. Intl.Collator.prototype.compare)
```
Comment (Huxpro) [inline doc/intl.md:5]:
```suggestion
- New functions or properties in the `Intl` object prototypes (e.g. `Intl.Collator.prototype.compare`)
```

File: doc/intl.md [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+This document describes the current state of Android implementation of Intl APIs(formally ECMA-402 specification) in Hermes JavaScript engine. ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and typically adds new capabilities typically as,
+- New Intl service constructors (for e.g. Intl.Collator, Intl.NumberFormat etc...) or extending existing ones by accepting more parameters
+- New functions or properties in the Intl object prototypes (for e.g. Intl.Collator.prototype.compare)
+- New locale aware functions in standard Javascript object prototypes (for e.g. String.prototype.localeCompare)
+
+One standard internationalization implementation strategy followed by other engines, is to bundle a library and data (typically [icu](http://site.icu-project.org/)) along with the application package. This guarantees deterministic functionality at the cost of applications package bloat.  We decided to consume the Android platform libraries and data for space efficiency, but at the cost of some variance in behaviour across  Android platforms.
```
Comment (Huxpro) [inline doc/intl.md:8]:
```suggestion
One standard internationalization implementation strategy followed by other engines, is to bundle a library and data (typically [ICU](http://site.icu-project.org/)) along with the application package. This guarantees deterministic functionality at the cost of applications package bloat. We decided to consume the Android platform libraries and data for space efficiency, but at the cost of some variance in behaviors across Android platforms.

```

File: doc/intl.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+- New Intl service constructors (for e.g. Intl.Collator, Intl.NumberFormat etc...) or extending existing ones by accepting more parameters
+- New functions or properties in the Intl object prototypes (for e.g. Intl.Collator.prototype.compare)
+- New locale aware functions in standard Javascript object prototypes (for e.g. String.prototype.localeCompare)
+
+One standard internationalization implementation strategy followed by other engines, is to bundle a library and data (typically [icu](http://site.icu-project.org/)) along with the application package. This guarantees deterministic functionality at the cost of applications package bloat.  We decided to consume the Android platform libraries and data for space efficiency, but at the cost of some variance in behaviour across  Android platforms.
+
+# ECMA-402 Compliance
```
Comment (Huxpro) [inline doc/intl.md:10]: ❤️

File: doc/intl.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+- New functions or properties in the Intl object prototypes (for e.g. Intl.Collator.prototype.compare)
+- New locale aware functions in standard Javascript object prototypes (for e.g. String.prototype.localeCompare)
+
+One standard internationalization implementation strategy followed by other engines, is to bundle a library and data (typically [icu](http://site.icu-project.org/)) along with the application package. This guarantees deterministic functionality at the cost of applications package bloat.  We decided to consume the Android platform libraries and data for space efficiency, but at the cost of some variance in behaviour across  Android platforms.
+
+# ECMA-402 Compliance
+
+## Supported
+
+- Intl.Collator
```
Comment (Huxpro) [inline doc/intl.md:14]: Please make sure those are backticks-quoted as well

File: doc/intl.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+
+- Intl.DateTimeFormat properties
+   - [dateStyle/timeStyle](https://tc39.es/proposal-intl-datetime-style/)
+   - [dayPeriod](https://github.com/tc39/ecma402/issues/29)
+   - [fractionalSecondDigits](https://github.com/tc39/ecma402/pull/347)
+- [BigInt.prototype.toLocaleString](https://tc39.es/ecma402/#sup-bigint.prototype.tolocalestring)
+
+## Excluded
+
+- [formatMatcher](https://tc39.es/ecma402/#sec-basicformatmatcher) property in Intl.DateTimeFormat.
```
Comment (Huxpro) [inline doc/intl.md:69]: Could you explain with more details like what you did in `*.docx`? You also mentioned that Firefox/Chrome does not seem to respect this.

File: doc/intl.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+
+## Known Issues
+
+### Android 11
+
+- The keys of the object returned by &#39;resolvedOptions&#39; function in all Intl services are not deterministically ordered as prescribed by spec.
+- DateFormat: ECMAScript [beginning of time](https://www.ecma-international.org/ecma-262/11.0/index.html#sec-time-values-and-time-range) (-8,640,000,000,000,000), is formatted as November 271817, instead of expected April 271822.
```
Comment (Huxpro) [inline doc/intl.md:76]: It's very important to properly format this sentence :) I'm assuming `April 271822` is the function output? They should be `-quoted

File: doc/intl.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+- Intl.NumberFormat implementation has some rough edges in supporting the following properties,
+  - style: 'unit'
+  - notation: 'compact'
+  - signDisplay
+  - currencyFormat
+
+### Android 10 (SDK<30) and older
```
Comment (Huxpro) [inline doc/intl.md:83]:
```suggestion
### Android 10 and older (SDK < 30)

```

File: doc/intl.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+
+### Android 7.0 - 7.1 (SDK<24) and older
+
+- Collator: Doesn&#39;t canonically decompose the input strings. Canonically equivalent string with non-identical code points may not match.
+- getCanonicalLocales: Unicode/CLDR version differences (Expected SameValue(«und-u-ca-ethiopic-amete-alem», «und-u-ca-ethioaa») to be true; Expected SameValue(«und-u-ks-primary», «und-u-ks-level1») to be true)
+- Unit style does not work.
+- There are issues in the precision configuration due to lack of APIs.
+- DateFormat: There are issues with the calendar configuration which needs to be dug into.
+
+### SDK<21 and older
```
Comment (Huxpro) [inline doc/intl.md:113]:
```suggestion
### SDK < 21
```

`<` says all

File: doc/intl.md [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+
+# Internationalization framework in Android Platform
+
+Our implementation is essentially projecting the Android platform provided internationalization facilities through the ECMA-402 specified services. It implies that the results of running the same code may vary between devices running different versions of Android.
+
+Android platform internationalization libraries have been based on [ICU4j project](https://unicode-org.github.io/icu-docs/#/icu4j). Version of ICU4j and the backing [CLDR data](http://cldr.unicode.org/) varies across Android platform versions. Also, the ICU APIs were never exposed directly, but only through wrappers and aliases. This results in significant variance in internationalization API surface and data across platform versions.
+
+The following table summarizes ICU, CLDR and Unicode versions available on the Android platforms that we intent to support.
+
+**Platform 24+ where ICU4j APIs are available.**
```
Comment (Huxpro) [inline doc/intl.md:125]: LOVE THIS TABLE

File: doc/intl.md [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+| --- | --- | --- | --- |
+| Android 11 (API level 30) | ICU4J 66.1 ([ref](https://android.googlesource.com/platform/external/icu/+/refs/heads/android11-mainline-release/icu4j/readme.html)) | Unicode 13 beta | CLDR 36.1 |
+| Android 10 (API level 29) | ICU4j 63.2 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 34 | Unicode 11.0 |
+| Android 9 (API level 28) | ICU4j 60.2 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 32.0.1 | Unicode 10.0 |
+| Android 8.0 - 8.1 (API levels 26 - 27) | ICU4j 58.2( [ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 30.0.3 | Unicode 9.0 |
+| Android 7.0 - 7.1 (API levels 24 - 25) | ICU4j 56 ([ref](https://developer.android.com/guide/topics/resources/internationalization))| CLDR 28 | Unicode 8.0 |
+
+<br />
+
+**Pre-24 platforms**
```
Comment (Huxpro) [inline doc/intl.md:136]:
```suggestion
### Pre-24 platforms
```

🤔   I haven't tried it locally but I imagined you can gain some proper margin by making it a header? (Same applies to the first table head)

File: doc/intl.md [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+
+| Android Platform Version | ICU | Unicode | CLDR
+| --- | --- | --- | --- |
+| Android 6.0 (API level 23) | ICU4j 55.1 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 27.0.1 | Unicode 7.0 |
+| Android 5.0 (API levels 21–22) | ICU4j 53 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 25 | Unicode 6.3 |
+| Android 4.4 (API levels 19–20) | ICU4j 51 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 23 | Unicode 6.2 |
+| Android 4.3 (API level 18) | ICU4j 50 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 22.1 | Unicode 6.2 |
+| Android 4.1 (API levels 16–17) | ICU4j 4.8 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 2.0 | Unicode 6.0 |  
+  
+<br />
```
Comment (Huxpro) [inline doc/intl.md:146]: 😅

File: doc/intl.md [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+To Summarize, there are three big partitions in the platform domain,
+
+Platforms <= 24 has much better internationalization support than earlier, as many ICU classes are available as is.
+
+Platforms 21-24 still has reasonable internationalization support, by allowing creation of Locale objects and exposing selected ICU services through java.text namespace.
+
+Platforms < 21 doesn't allow creation of Locale objects from tags, severely limiting general purpose international code.
+
+
+# Impact on Application Size
```
Comment (Huxpro) [inline doc/intl.md:157]: Yeah we should include this 👍

File: doc/intl.md [hunk 18]
```diff
@@ -0,0 +1,192 @@
...
+
+The following numbers are measured using a test application which takes dependency on the Hermes library to evaluate a JavaScript snippet. Essentially, enabling Intl APIs adds 57-62K per ABI.
+
+| **Product APK Size** | **NOINTL** | **INTL** | **DIFF** | **PERC** |
```
Comment (Huxpro) [inline doc/intl.md:161]: Awesome table!

**Review by Huxpro — CHANGES_REQUESTED** [review]
LOVE THE CONTENT. 

Most of my reviews are nits regarding formatting. I understood that copying from `docx` can lose a lot of information.

I tend to think we should consider making this richer enough that it can replace the `.docx` eventually. `.md` is much more accessible from Github than `.docx` and can be hosted to the Hermes website as well ;)

File: doc/intl.md [hunk 19]
```diff
@@ -0,0 +1,196 @@
...
+---
+
+This document describes the current state of Android implementation of the [ECMAScript Internationalization API Specification](https://tc39.es/ecma402/) (ECMA-402, or `Intl`). ECMA-402 is still evolving and the latest iteration is [7th edition](https://402.ecma-international.org/7.0/) which was published in June 2020. Each new edition is built on top of the last one and adds new capabilities typically as,
+- New `Intl` service constructors (e.g. `Intl.Collator`, `Intl.NumberFormat` etc.) or extending existing ones by accepting more parameters
+- New functions or properties in `Intl` objects (e.g. `Intl.Collator.prototype.compare`)
+- New locale aware functions in standard Javascript object prototypes (e.g. `String.prototype.localeCompare`)
+
+One popular implementation strategy followed by other engines, is to bundle an internationalization framework (typically [ICU](http://site.icu-project.org/)) along with the application package. This guarantees deterministic behaviours at the cost of applications package bloat. We decided to consume the Android platform provided facilities for space efficiency, but at the cost of some variance in behaviours across Android platforms.
```
Comment (Huxpro) [inline doc/intl.md:11]: Love the new contents 👍

File: doc/intl.md [hunk 20]
```diff
@@ -0,0 +1,196 @@
...
+- `Intl.NumberFormat` implementation has some rough edges in supporting the following properties,
+  - `style`: 'unit'
+  - `notation`: 'compact'
+  - `signDisplay`
+  - `currencyFormat`
+
+### Android 10 and older (SDK < 30)
+
+- `Intl.NumberFormat`: Scientific notation formatting has issues on some cases such as Infinity (e.g. Expected SameValue(«-∞E0», «-∞») to be true).
+- `Intl.NumberFormat`: Compact notation `formatToParts` doesn't identify unit, hence we report unit as 'literal' (Compact short: 987654321: parts[1].type Expected SameValue(«literal», «compact») to be true)
```
Comment (Huxpro) [inline doc/intl.md:89]: I don't actually understand whatever in the `(Compact short: ...)`. The `SameValue` part looks like the spec text but the `«...»` looks like some console output from other engines? I'm not sure. Could you make this more readable?

File: doc/intl.md [hunk 21]
```diff
@@ -0,0 +1,196 @@
...
+- `Intl.NumberFormat`: `formatToParts` function doesn't produce expected result with Scientific/Engineering notation and input Infinity (`-Infinity` - engineering: length Expected SameValue(«4», «2») to be true).
```
Comment (Huxpro) [inline doc/intl.md:90]: Same here, what is "`-Infinity` - engineering:"?

File: doc/intl.md [hunk 22]
```diff
@@ -0,0 +1,196 @@
...
+- DateFormat: ECMAScript [beginning of time](https://www.ecma-international.org/ecma-262/11.0/index.html#sec-time-values-and-time-range) (-8,640,000,000,000,000), is formatted as `November 271817`, instead of expected `April 271822`.
+- `Intl.NumberFormat` implementation has some rough edges in supporting the following properties,
+  - `style`: 'unit'
+  - `notation`: 'compact'
+  - `signDisplay`
+  - `currencyFormat`
+
+### Android 10 and older (SDK < 30)
+
+- `Intl.NumberFormat`: Scientific notation formatting has issues on some cases such as Infinity (e.g. Expected SameValue(«-∞E0», «-∞») to be true).
```
Comment (Huxpro) [inline doc/intl.md:88]:
```suggestion
- `Intl.NumberFormat`: Scientific notation formatting has issues on some cases such as Infinity (e.g. Expected `SameValue(«-∞E0», «-∞»)` to be true).
```

I don't know...are they code?

File: doc/intl.md [hunk 23]
```diff
@@ -0,0 +1,196 @@
...
+| --- | --- | --- | --- |
+| Android 6.0 (API level 23) | ICU4j 55.1 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 27.0.1 | Unicode 7.0 |
+| Android 5.0 (API levels 21–22) | ICU4j 53 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 25 | Unicode 6.3 |
+| Android 4.4 (API levels 19–20) | ICU4j 51 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 23 | Unicode 6.2 |
+| Android 4.3 (API level 18) | ICU4j 50 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 22.1 | Unicode 6.2 |
+| Android 4.1 (API levels 16–17) | ICU4j 4.8 ([ref](https://developer.android.com/guide/topics/resources/internationalization)) | CLDR 2.0 | Unicode 6.0 |  
+  
+<br />
+
+In summary,
```
Comment (Huxpro) [inline doc/intl.md:151]: Maybe this parts should move to the beginning of the "Known issues" sections?

File: doc/intl.md [hunk 24]
```diff
@@ -0,0 +1,196 @@
...
+
+Platforms >= 24 have much better internationalization support than earlier, as many ICU classes are available as is.
```
Comment (Huxpro) [inline doc/intl.md:153]: Maybe an ordered list?
- `30` blah
- `>= 24` blah
- `21-24` blah
- `< 21` blah

File: doc/intl.md [hunk 25]
```diff
@@ -0,0 +1,196 @@
...
+   - [`dateStyle/timeStyle`](https://tc39.es/proposal-intl-datetime-style/)
+   - [`dayPeriod`](https://github.com/tc39/ecma402/issues/29)
+   - [`fractionalSecondDigits`](https://github.com/tc39/ecma402/pull/347)
+- [`BigInt.prototype.toLocaleString`](https://tc39.es/ecma402/#sup-bigint.prototype.tolocalestring)
+
+## Excluded
+
+- `Intl.DateTimeFormat`: [`formatMatcher`](https://tc39.es/ecma402/#sec-basicformatmatcher) parameter is not respected. The parameter enables the implementation to pick the best display format when it supports only a subset of all possible formats. ICU library in Android platform and hence our implementation allows all subsets and formats which makes this `formatMatcher` property unnecessary.
+
+## Known Issues
```
Comment (Huxpro) [inline doc/intl.md:74]:
```suggestion
## Known Issues across Android SDKs
```

Or maybe "Variance across Android SDKs" or "Limitations across Android SDKs" since we are not going to fix them (because of the platform limitations), right?

**Review by Huxpro — COMMENTED** [review]
Awesome! Only a few more questions and formatting recommendation!

I'm ready to merge and publish this 🤩

**Review by dulinriley — APPROVED** [review]
Thanks for adding the sidebars to the website! I'll let @Huxpro finish this review

**Review by Huxpro — CHANGES_REQUESTED** [review]
I've unresolved some comments that I believe you did not notice from my 2nd round of comments after your commits to address the 1st round of review.

File: doc/intl.md (hunk 21 above)
Comment (mganandraj) [reply doc/intl.md:90]: This one is strange .. When i was testing a few months back, `-Inifinity` was foamtted as something like `∞E0` on older than platform 30 .. but from a quick test, it doesn't seem to be the case anymore .. May be the icu libraries are getting updated over the air .. I'll keep the entry anyways but make the text clearer.

File: doc/intl.md (hunk 20 above)
Comment (mganandraj) [reply doc/intl.md:89]: I've copied out the assert failure statement as it .. my bad :) I'm making the text better readable.

Also, please note that I tried to document the test code with these failures. For instance take a look at : https://github.com/facebook/hermes/blob/master/android/intltest/java/com/facebook/hermes/test/HermesIntlNumberFormatTest.java

File: doc/intl.md (hunk 23 above)
Comment (mganandraj) [reply doc/intl.md:151]: Xuan.. It makes sense to move it  there .. But as it is a summary of the library support on each platform, it make sense to keep it here as well .. It is a tough one .. Shall i keep it here as it for now being ?

File: doc/intl.md (hunk 25 above)
Comment (mganandraj) [reply doc/intl.md:74]: Makese sense ..

File: doc/intl.md (hunk 23 above)
Comment (Huxpro) [reply doc/intl.md:151]: It's a "maybe" suggestion so I'm fine whatever you choose ;)

File: doc/intl.md (hunk 20 above)
Comment (Huxpro) [reply doc/intl.md:89]: Oh, so it's error from test262 (well, test402 I guess). I see.

Since you already explained them in prose. Maybe we can just link to those tests if people are interested in looking at the exact edge cases.

File: doc/intl.md (hunk 8 above)
Comment (mganandraj) [reply doc/intl.md:13]: Ah .. I thought about your question last week .. but missed actually replying to it ! 

I am hoping we will continue to keep the docx as a richer detailed documentation for the Intl developers and the markdown as the quick reference for Intl consumers .. I see that many aspects of our representations in the docx can be useful for the consumers too, but will also also make the contents complicated .. I think we could work on it as part of our next wave of Intl feature developement ..

File: doc/intl.md [hunk 26]
```diff
@@ -0,0 +1,195 @@
...
+- `Intl.NumberFormat`: CompactFormatter doesn't respect the precision inputs (Expected SameValue(«9,900만», «9877만») to be true; Expected SameValue(«990M», «988M») to be true).
+
+### Android 7.0 - 7.1 and older (SDK < 26)
+
+- `Intl.getCanonicalLocales`: Unicode/CLDR version differences Expected SameValue(«und-u-ms-imperial», «und-u-ms-uksystem») to be true
+
+### Android 7.0 - 7.1 and older (SDK < 24)
+
+- `Intl.Collator`: Doesn't canonically decompose the input strings. Canonically equivalent string with non-identical code points may not match.
+- `Intl.getCanonicalLocales`: Unicode/CLDR version differences (Expected SameValue(«und-u-ca-ethiopic-amete-alem», «und-u-ca-ethioaa») to be true; Expected SameValue(«und-u-ks-primary», «und-u-ks-level1») to be true)
```
Comment (Huxpro) [inline doc/intl.md:110]: There are still some `Expected SameValue` here that might confuse our reader

**Review by Huxpro — APPROVED** [review]
Thanks! 

I'm going to approve it and start with importing this. But if you got a chance, maybe consider improve those `Expected SameValue`

File: doc/intl.md [hunk 27]
```diff
@@ -0,0 +1,195 @@
...
+- `Intl.NumberFormat` implementation has some rough edges in supporting the following properties,
+  - `style`: 'unit'
+  - `notation`: 'compact'
+  - `signDisplay`
+  - `currencyFormat`
+
+### Android 10 and older (SDK < 30)
+
+- `Intl.NumberFormat`: Scientific notation formatting has issues on some cases. e.g. `-Infinity` may get formatted as '-∞E0' instead of expected '-∞'. Another manifestation of the issues is that the formatToParts may return 4 parts instead of 2.
+- `Intl.NumberFormat`: Compact notation `formatToParts` doesn't identify unit, hence we report unit as 'literal'. For e.g. the second part of "100ac" gets reported as "literal" instead of "compact"
```
Comment (Huxpro) [inline doc/intl.md:89]: I really appreciate this example. Love it!
